r2-shared/Publication/PublicationManifest.ts

```diff
@@ -0,0 +1,196 @@
+import Metadata from "./Metadata";
+import Store from "../Store/Store";
+import { Link, Links } from "./Link";
+
+export default class PublicationManifest {
+  public readonly context: Array<string>;
+  public readonly metadata: Metadata;
+  public readonly links: Links;
+  public readonly readingOrder: Links;
+  public readonly resources: Links;
+  public readonly tableOfContents: Links;
+
+  public readonly manifestUrl: URL;
```

Comment:
`manifestUrl` only works for WebPub right? Unless this always points to the publication URL no matter its type. What is it used for?

Reply:
Ah yeah so every review comment is kinda tied to discussing what happens there. 

I started from R2D2BC as I quickly noticed there was a massive overlap with NYPL original code + ours so it definitely needs feedback from other parties since it’s been probably used in a lot of places.

Reply:
Okay we can discuss about that on the next call.

Note that it's important that `manifest` is private in `Publication`, so all these functions (if we keep them) should probably be moved to `Publication` instead of `Manifest` (which doesn't exist in R2D2BC)

Reply:
Probably shouldn't be here in the first place, R2D2BC doesn't include the entire shared model as it has been defined way after R2D2BC has been developed, R2D2BC only includes the Publication and Locator model, as created before a specification was done. This models here are not to be compared or even mixed up with the R2D2BC models, but rather be created as a "new" shared models that can be added or replaces in previous implementations that haven't refactored yet to the newest definitions.

Reply:
Thanks for the confirmation, this has been removed.

Reply:
Great, that will simplify things 👍